### PR TITLE
Added auto label for issues

### PR DESCRIPTION
**Pull request by muthuannamalai12**
Added auto label issues script which labels issues with gssoc21 when contributor asks to work under it.

### CODE REVIEW

**Comment by paramsiddharth** [comment]
Would it work for "GSSoC'21", with the apostrophe?

**Comment by muthuannamalai12** [comment]
> Would it work for "GSSoC'21", with the apostrophe?

It works with GSSoC21 but not with an apostrophe in between

**Comment by paramsiddharth** [comment]
> > Would it work for "GSSoC'21", with the apostrophe?
> 
> It works with GSSoC21 but not with an apostrophe in between

Please update it to support that.

**Comment by muthuannamalai12** [comment]
> > > Would it work for "GSSoC'21", with the apostrophe?
> > 
> > 
> > It works with GSSoC21 but not with an apostrophe in between
> 
> Please update it to support that.

The GitHub action I am using is not supporting apostrophe.

**Comment by paramsiddharth** [comment]
> > > > Would it work for "GSSoC'21", with the apostrophe?
> > > 
> > > 
> > > It works with GSSoC21 but not with an apostrophe in between
> > 
> > 
> > Please update it to support that.
> 
> The GitHub action I am using is not supporting apostrophe.

Update it then. It would be fairly easy. You need to add the labels containing an apostrophe to the `labels-synonyms`. In addition to that, you might want to add synonyms like `GirlScript Summer of Code` to the list too. That would ensure no synonymous patterns are left unmatched.

**Comment by paramsiddharth** [comment]
@muthuannamalai12 Are you able to do it?

**Comment by muthuannamalai12** [comment]
> @muthuannamalai12 Are you able to do it? Are you facing any problems?

I have tried it by adding in label synonyms but it give erros and it is not working.

**Comment by paramsiddharth** [comment]
>  In single quoted strings the single quote character can be escaped by prefixing it with another single quote, basically doubling it.

Source: [A YAML Primer](https://docs.octoprint.org/en/master/configuration/yaml.html#:~:text=In%20single%20quoted%20strings%20the,just%20indent%20the%20following%20lines.)

@muthuannamalai12 Since the outermost is a single quote, try using the abovementioned technique to escape single-quotes.

File: .github/workflows/issue.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  issues:
+      types: ['opened']
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Renato66/auto-label@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          labels-synonyms: '{"GSSOC21":["gssoc","gssoc21","GSSOC21","GSSOC","GSSoC21"]}'
```
Comment (paramsiddharth) [inline .github/workflows/issue.yml:12]: Try making it the following:
```yaml
labels-synonyms: '{"GSSOC21":["gssoc","gssoc21","GSSOC21","GSSOC","GSSoC''21"]}'
```
Test it and confirm if it works.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

File: .github/workflows/issue.yml (hunk 1 above)
Comment (muthuannamalai12) [reply .github/workflows/issue.yml]: > Try making it the following:
> 
> ```yaml
> labels-synonyms: '{"GSSOC21":["gssoc","gssoc21","GSSOC21","GSSOC","GSSoC''21"]}'
> ```
> 
> Test it and confirm if it works.

It works absolutely fine. Thank you for the information

**Comment by paramsiddharth** [comment]
Good job!

**Pull request merged**